### ISSUE TITLE
Add `ack --count` for counting barriers

### ISSUE THREAD

**Issue by hershal** [hershal/synchronize-cli#1]
Ack needs a counting barrier. This would look like:

```
$ ack <topic> --count 4 && ./do-something.sh
```

Which would require 4 `syn <topic>` to unlock. Also consider adding `syn --count 2 <topic>` to send 2 syns to that ack server.

**Comment by hershal** [comment]
Might have to do a leader election here on ack servers. Suppose we have multiple ack servers listening for different counts:

```
ack --count 1
```
and 
```
ack --count 2
```

if we send a `syn`, then the first ack (the server) will try to exit but can't because it has the second ack (the client) listening for requests. We'll have to do an election here to make the ack with the highest count the leader and then restart the server and force clients to reconnect. 

We have this problem today when a leading ack server exits when there are multiple clients. We need to always know who the leader is and who the other clients are so that the clients can reconfigure themselves and reconnect to the right server when it restarts.

**Comment by hershal** [comment]
This is fixed as of 32e91dafcb7df96be73080b6a191b4b0f3e1730d
